source/funkin/editors/character/CharacterConfig.hx

```diff
@@ -0,0 +1,169 @@
+			xml = new Access(charXML);
+		}
+
+		charData = xmlToData(charData, xml);
+
+		return charData;
+	}
+
+	public static function xmlToData(charData:CharacterData, xml:Access):CharacterData
+	{
+		if (xml.has.isPlayer)
+			charData.isPlayer = (xml.att.isPlayer == "true");
+		if (xml.has.isGF)
+			charData.isGF = (xml.att.isGF == "true");
+		if (xml.has.x)
+			charData.offsetX = Std.parseFloat(xml.att.x);
+		if (xml.has.y)
+			charData.offsetY = Std.parseFloat(xml.att.y);
+		if (xml.has.gameOverChar)
+			charData.gameOverChar = xml.att.gameOverChar;
+		if (xml.has.camx)
+			charData.camOffsetX = Std.parseFloat(xml.att.camx);
+		if (xml.has.camy)
+			charData.camOffsetY = Std.parseFloat(xml.att.camy);
+		if (xml.has.holdTime)
+			charData.holdTime = CoolUtil.getDefault(Std.parseFloat(xml.att.holdTime), 4);
+		if (xml.has.flipX)
+			charData.flipX = (xml.att.flipX == "true");
+		if (xml.has.icon)
+			charData.icon = xml.att.icon;
+		if (xml.has.iconColor)
+				charData.iconColor = stringToRGB(xml.att.iconColor);
+		if (xml.has.scale)
+		{
+			var scale = Std.parseFloat(xml.att.scale).getDefault(1);
+			charData.scale = scale;
+		}
+		if (xml.has.antialiasing)
+			charData.antialiasing = (xml.att.antialiasing == "true");
+		if (xml.has.sprite)
+			charData.sprite = xml.att.sprite;
+		else
+			charData.sprite = charData.name;
+
+		for (anim in xml.nodes.anim)
+		{
+			var animType:XMLAnimType = NONE;
+			var animloop = false;
+			if (anim.has.loop)
+				animloop = (anim.att.loop == "true");
+			charData.animations.push(XMLUtil.extractAnimFromXML(anim, animType, animloop));
+		}
+		return charData;
+	}
+
+	public static function dataToXml(charData:CharacterData):Access
+	{
+		var root = Xml.createElement("character");
+		root.set("x", Std.string(Math.floor(charData.offsetX)));
+		root.set("y", Std.string(Math.floor(charData.offsetY)));
+		root.set("gameOverChar", Std.string(charData.gameOverChar));
+		root.set("camx", Std.string(Math.floor(charData.camOffsetX)));
+		root.set("camy", Std.string(Math.floor(charData.camOffsetY)));
+		root.set("holdTime", Std.string(charData.holdTime));
+		root.set("flipX", Std.string(charData.flipX));
+		root.set("icon", charData.icon);
+		root.set("iconColor", Std.string(charData.iconColor));
+		root.set("scale", Std.string(charData.scale));
+		root.set("antialiasing", Std.string(charData.antialiasing));
+		root.set("sprite", Std.string(charData.sprite));
+		root.set("isPlayer", Std.string(charData.isPlayer));
+		root.set("isGF", Std.string(charData.isGF));
+
+		for (anim in charData.animations)
+		{
+			var child:Xml = Xml.createElement('anim');
+			child.set("name", Std.string(anim.name));
+			child.set("anim", Std.string(anim.anim));
+			child.set("loop", Std.string(anim.loop));
+			child.set("fps", Std.string(anim.fps));
+			child.set("x", Std.string(anim.x));
+			child.set("y", Std.string(anim.y));
+			child.set("indices", Std.string(anim.indices));
+			root.addChild(child);
```

Comment:
if you can, make it so it doesnt include properties that are the default, like if holdTime != 4 then add it, and so on, resulting in smaller xml files

Reply:
Every default value has been set by when the character gets it's own characterData.

Reply:
im saying like so it doesnt generate it automatically in the saved xml